strutil/shlex/shlex_test.go

```diff
@@ -157,3 +157,69 @@ func TestNastyReader(t *testing.T) {
 		t.Errorf("unexpected error")
 	}
 }
+
+func TestQuote(t *testing.T) {
+	safeUnquoted := "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ" + "0123456789" + "@%_-+=:,./"
+	unicodeSample := "\xe9\xe0\xdf" // e + acute accent, a + grave, sharp s
+	unsafe := "\"`$\\!" + unicodeSample
+
+	type testInfo struct {
+		input    string
+		expected string
+	}
+
+	quoteTests := []testInfo{{
+		input:    ``,
+		expected: `''`,
+	}, {
+		input:    safeUnquoted,
+		expected: safeUnquoted,
+	}, {
+		input:    `foo bar xyz`,
+		expected: `'foo bar xyz'`,
+	}}
+
+	for _, u := range unsafe {
+		quoteTests = append(quoteTests, testInfo{
+			input:    `foo` + string(u) + `bar`,
+			expected: `'foo` + string(u) + `bar'`,
+		})
+		quoteTests = append(quoteTests, testInfo{
+			input:    `foo` + string(u) + `'bar'`,
+			expected: `'foo` + string(u) + `'"'"'bar'"'"''`,
+		})
+	}
+
+	for _, test := range quoteTests {
+		quoted := Quote(test.input)
+		if quoted != test.expected {
+			t.Errorf("expected %s, got %s", test.expected, quoted)
```

Comment:
With table-driven tests it's best to have some way of knowing which test that failed. It's usually best to use `t.Run()` inside the loop and explicitly give the test a name in your struct, like

```go
quoteTests := []testInfo{{
		name:     "empty",
		input:    ``,
		expected: `''`,
	}, {
	...
```

Or you could avoid the explicit name and make the `t.Run` name `fmt.Sprintf("Quote(%q)", test.input)`.